react/components/SKUSelector/components/Variation.tsx

```diff
@@ -60,21 +59,19 @@ const Variation: FC<Props> = ({
   containerClasses: containerClassesProp,
 }) => {
   const { name, options } = variation
-  const [showAll, setShowAll] = useState(false)
+  
+  const [showAll, setShowAll] = useState(() => {
+    const selectedOptionPosition = findSelectedOption(selectedItem)(options)
+    return selectedOptionPosition >= visibleItemsWhenCollapsed
+  })
```

Comment:
How does this work? Because you're passing a function to the first value to `showAll`, but then you set `showAll` to be a boolean, I didn't get it why this should be a function

Reply:
Opa, se você passa uma função ele faz um lazy evaluation e essa função só é chamada no primeiro render, é para inicializar o estado somente, procura na doc de react de useState